awx/api/serializers.py

```diff
@@ -3333,30 +2949,12 @@ def get_summary_fields(self, obj):
                     'kind': cred.kind,
                     'cloud': cred.credential_type.kind == 'cloud'
                 }
-                if self.version > 1:
-                    summarized_cred['credential_type_id'] = cred.credential_type_id
                 all_creds.append(summarized_cred)
                 if cred.credential_type.kind in ('cloud', 'net'):
                     extra_creds.append(summarized_cred)
-                elif summarized_cred['kind'] == 'ssh':
-                    credential = summarized_cred
-                elif summarized_cred['kind'] == 'vault':
-                    vault_credential = summarized_cred
-        # Selectively apply those fields, depending on view deetails
-        if (self.is_detail_view or self.version == 1) and credential:
-            summary_fields['credential'] = credential
-        else:
-            # Credential could be an empty dictionary in this case
-            summary_fields.pop('credential', None)
-        if (self.is_detail_view or self.version == 1) and vault_credential:
-            summary_fields['vault_credential'] = vault_credential
-        else:
-            # vault credential could be empty dictionary
-            summary_fields.pop('vault_credential', None)
-        if self.version > 1:
-            if self.is_detail_view:
-                summary_fields['extra_credentials'] = extra_creds
-            summary_fields['credentials'] = all_creds
+        if self.is_detail_view:
+            summary_fields['extra_credentials'] = extra_creds
```

Comment:
Since you removed the `credential` field, I don't know if I agree with providing `extra_credentials` in the summary fields (I know there's a lot of special handling in the job launch view too). Aren't these both parts of the same compatibility shim?

I do see `extra_credentials` referenced in javascript, which does seem like a problem.

So I'm still lacking an idea of exactly which fields are being removed, and why.

Reply:
hmm... going back and fourth with the job launch view, I think I'm seeing better the intentionality of this. I'm inferring that `extra_credentials` was maybe introduced later than `credential` and `vault_credential`, and gives cloud credentials in general.

Reply:
I've kind of gone back and worth on this, and my rationale here was that the `extra_credentials` interface is considerably _more recent_ than the the `credential` and `vault_credential` polyfills (cir. Ansible Tower 3.3 vs Ansible Tower 3.2).